server/src/main/java/org/opensearch/action/admin/indices/settings/put/TransportUpdateSettingsAction.java

```diff
@@ -145,10 +147,10 @@ protected ClusterBlockException checkBlock(UpdateSettingsRequest request, Cluste
             }
         }
 
+        final String[] requestIndexNames = Arrays.stream(requestIndices).map(Index::getName).toArray(String[]::new);
         return allowSearchableSnapshotSettingsUpdate
             ? null
-            : state.blocks()
-                .indicesBlockedException(ClusterBlockLevel.METADATA_WRITE, indexNameExpressionResolver.concreteIndexNames(state, request));
+            : state.blocks().indicesBlockedException(ClusterBlockLevel.METADATA_WRITE, requestIndexNames);
```

Comment:
Why is this change here? How is this related to adding support for `number_of_replicas`?

Reply:
I just think we can reuse the first result to avoid computing request indices twice. This has nothing to do with adding support for `number_of_replicas`